src/filters/index.ts

```diff
@@ -0,0 +1,57 @@
+import type { Repo } from "~common/repo";
+import NpmFilter from "./npm";
+
+const filters: { [key: string]: RepoFilter } = {
+  npm: NpmFilter,
+}
+
+const defaultFilter = (repo: Repo) => {
+  return []
+}
+
+type FilterFn = typeof defaultFilter
+
+export type RepoFilter = {
+  name: string;
+  searchTerms: string[];
+  fn: FilterFn;
+}
+
+export type FilterMatchHeuristic = {
+  name: string;
+  dataPoints: any[];
+}
+
+const prepareSearch = (repo: Repo) => {
+  for (const filterName in filters) {
+    if (Object.prototype.hasOwnProperty.call(filters, filterName)) {
+      const filter = filters[filterName];
+
+      repo.search.addSearchTerms(filter.searchTerms);
+    }
+  }
+}
+
+export const calcHeuristics = async (repo: Repo): Promise<FilterMatchHeuristic[]> => {
+  // Prepare search related filterings
+  prepareSearch(repo);
+
+  // Generate Repo analysis based on filters
+  await repo.analyze();
+
+  // Gather heuristics per filter
+  let heuristics: FilterMatchHeuristic[] = [];
+
+  for (const filterName in filters) {
+    if (Object.prototype.hasOwnProperty.call(filters, filterName)) {
```

Comment:
If we're looping through filterName within filters, why do we need to check it exists?